File: php/features.php
<?php
declare(strict_types=1);

// JikanRest SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class JikanRestFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new JikanRestBaseFeature();
            case "test":
                return new JikanRestTestFeature();
            default:
                return new JikanRestBaseFeature();
        }
    }
}

File: py/features.py
# JikanRest SDK feature factory

from feature.base_feature import JikanRestBaseFeature
from feature.test_feature import JikanRestTestFeature


def _make_feature(name):
    features = {
        "base": lambda: JikanRestBaseFeature(),
        "test": lambda: JikanRestTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
